### PR TITLE
fasttree: replace gcc dependency with libomp

### PR DESCRIPTION
**Pull request by SeekingMeaning**
- [ ] Have you followed the [guidelines for contributing](https://github.com/brewsci/homebrew-bio/blob/master/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/brewsci/homebrew-bio/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source FORMULA`, where `FORMULA` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict FORMULA` (after doing `brew install FORMULA`)?

-----

Formulae that might need to be bumped:

- pirate
- parsnp
- nullarbor
- roary

### CODE REVIEW

File: Formula/fasttree.rb [hunk 1]
```diff
@@ -21,13 +22,14 @@ class Fasttree < Formula
...
-    depends_on "gcc" if OS.mac? # needs openmp
+    on_macos do
+      depends_on "libomp"
+    end
   end
 
   def install
     opts = %w[-O3 -finline-functions -funroll-loops]
-    opts << "-DOPENMP" << "-fopenmp" if build.with? "openmp"
+    opts << "-DOPENMP" << "-L#{Formula["libomp"].opt_lib}" << "-lomp" if build.with? "openmp"
```
Comment (sjackman) [inline Formula/fasttree.rb:32]: `libomp` shouldn't be used on Linux.

File: Formula/fasttree.rb [hunk 2]
```diff
@@ -21,13 +22,14 @@ class Fasttree < Formula
   option "without-sse", "Disable SSE parallel instructions"
```
Comment (sjackman) [inline Formula/fasttree.rb:22]: You can remove these options if you like.

File: Formula/fasttree.rb [hunk 3]
```diff
@@ -21,13 +22,14 @@ class Fasttree < Formula
...
 
   if build.with? "openmp"
-    fails_with :clang # needs openmp
-    depends_on "gcc" if OS.mac? # needs openmp
+    on_macos do
+      depends_on "libomp"
+    end
```
Comment (sjackman) [inline Formula/fasttree.rb:27]: `on_macos` should go on the outside of the `if`.

File: Formula/fasttree.rb (hunk 1 above)
Comment (SeekingMeaning) [reply Formula/fasttree.rb:32]: Oh yeah whoops

File: Formula/fasttree.rb (hunk 3 above)
Comment (sjackman) [reply Formula/fasttree.rb:27]: You can remove the `if build.with?` though if you remove the options.

File: Formula/fasttree.rb [hunk 4]
```diff
@@ -16,20 +17,24 @@ class Fasttree < Formula
...
-    opts << "-DOPENMP" << "-fopenmp" if build.with? "openmp"
-    opts << "-DUSE_DOUBLE" if build.with? "double"
-    opts << "-DNO_SSE" if build.without? "sse"
+    opts = %w[
+      -O3
+      -finline-functions
+      -funroll-loops
+      -DOPENMP
+      -DUSE_DOUBLE
+      -DNO_SSE
```
Comment (sjackman) [inline Formula/fasttree.rb:31]:
```ruby
opts << "-DNO_SSE" if build.without? "sse"
```

```suggestion
```

`-DNO_SSE` was not used by default previously

Comment (SeekingMeaning) [reply Formula/fasttree.rb:31]: Oh whoops

**Comment by SeekingMeaning** [comment]
What should we do for broken dependents?

**Comment by sjackman** [comment]
```
==> brew test --verbose brewsci/bio/roary
118
==> FAILED
119
Testing brewsci/bio/roary
120
==> /home/linuxbrew/.linuxbrew/Cellar/roary/3.13.0/bin/roary -w 2>&1
121
Util.c: loadable library and perl binaries are mismatched (got handshake key 0xcd00080, needed 0xed00080)
122
Error: brewsci/bio/roary: failed
```
https://github.com/brewsci/homebrew-bio/pull/1123/checks?check_run_id=908545358#step:5:121

In a separate PR bump `roary` for `perl`.

I have a concept that I call "long shelf life bottles". Basically bumping a formulae in Homebrew/core should not cause breakage in bottles of third party taps. I'd really like this to be a feature of Homebrew 3. Happy to discuss more in Slack. Please record all the formulae that caused breakage in Brewsci/bio. `gcc`, `perl`, `gsl`, …

**Comment by sjackman** [comment]
```
==> brew test --verbose brewsci/bio/parsnp
113
==> FAILED
114
Testing brewsci/bio/parsnp
115
/usr/bin/sandbox-exec -f /private/tmp/homebrew20200724-17982-1eg180e.sb ruby -W0 -I $LOAD_PATH -- /usr/local/Homebrew/Library/Homebrew/test.rb /usr/local/Homebrew/Library/Taps/brewsci/homebrew-bio/Formula/parsnp.rb --verbose
116
==> /usr/local/Cellar/parsnp/1.2_3/bin/parsnp -v 2>&1
117
sh: line 1: 17995 Abort trap: 6           /usr/local/Cellar/parsnp/1.2_3/bin/parsnp -v 2>&1
118
dyld: Library not loaded: /usr/local/opt/gcc/lib/gcc/9/libgomp.1.dylib
119
  Referenced from: /usr/local/Cellar/parsnp/1.2_3/bin/parsnp
120
  Reason: image not found
121
Error: brewsci/bio/parsnp: failed
```
https://github.com/brewsci/homebrew-bio/pull/1123/checks?check_run_id=908545349#step:4:118

Merge the `parsnp` PR first, then rebase this PR onto `origin/develop`.

File: Formula/fasttree.rb (hunk 4 above)
Comment (sjackman) [reply Formula/fasttree.rb:31]: This comment still applies.

Comment (SeekingMeaning) [reply Formula/fasttree.rb:31]: Strange I was sure I had made this change

**Pull request merged**